tasks/discover-granules/tests/index.js

```diff
@@ -116,60 +115,150 @@ test('discover granules using HTTP', async (t) => {
       t.pass('Ignoring this test. Remote host seems to be down.');
     } else throw e;
   }
 });
 
-const discoverGranulesUsingS3 = (configure) => async (t) => {
-  const { event, event: { config } } = t.context;
-  // State sample files
-  const files = [
-    'granule-1.nc', 'granule-1.nc.md5',
-    'granule-2.nc', 'granule-2.nc.md5',
-    'granule-3.nc', 'granule-3.nc.md5'
-  ];
+const discoverGranulesUsingS3 = (configure, assert = assertDiscoveredGranules) =>
+  async (t) => {
+    const { event, event: { config } } = t.context;
+    // State sample files
+    const files = [
+      'granule-1.nc', 'granule-1.nc.md5',
+      'granule-2.nc', 'granule-2.nc.md5',
+      'granule-3.nc', 'granule-3.nc.md5'
+    ];
 
-  config.sourceBucketName = randomString();
-  config.collection.provider_path = randomString();
+    config.sourceBucketName = randomString();
+    config.collection.provider_path = randomString();
 
-  configure(config);
+    configure(t);
 
-  await validateConfig(t, config);
-  await s3().createBucket({ Bucket: config.sourceBucketName }).promise();
+    validateConfig(t, config);
+    await s3().createBucket({ Bucket: config.sourceBucketName }).promise();
 
-  try {
-    await Promise.all(files.map((file) =>
-      s3().putObject({
-        Bucket: config.sourceBucketName,
-        Key: `${config.collection.provider_path}/${file}`,
-        Body: `This is ${file}`
-      }).promise()));
-    await assertDiscoveredGranules(t, await discoverGranules(event));
-  } finally {
-    // Clean up
-    await recursivelyDeleteS3Bucket(config.sourceBucketName);
-  }
-};
+    try {
+      await Promise.all(files.map((file) =>
+        s3().putObject({
+          Bucket: config.sourceBucketName,
+          Key: `${config.collection.provider_path}/${file}`,
+          Body: `This is ${file}`
+        }).promise()));
+      assert(t, await discoverGranules(event));
+    } finally {
+      // Clean up
+      await recursivelyDeleteS3Bucket(config.sourceBucketName);
+    }
+  };
 
 test('discover granules using S3',
-  discoverGranulesUsingS3((config) => {
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    config.provider = {
+      id: 'MODAPS',
+      protocol: 's3',
+      host: config.sourceBucketName
+    };
+  }));
+
+test('discover granules without collection files config using S3',
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    // Without files config we should still discover granules, but the
+    // discovered granules will have empty files arrays.
+    config.collection.files = [];
+    config.provider = {
+      id: 'MODAPS',
+      protocol: 's3',
+      host: config.sourceBucketName
+    };
+  }, (t, output) => {
+    validateOutput(t, output);
+    t.is(output.granules.length, 3);
+    output.granules.forEach(({ files }) => t.is(files.length, 0));
+  }));
+
+test('discover granules without collection files config, but configuring collection to ignore it, using S3',
```

Comment:
If I remember correctly, the DAAC has collection files config, but choose to ignore them for discovering granules, and the files config is still used for other processing.
Maybe `ignoreFilesConfigForDiscovering` is more appropriate?

Reply:
Sure, makes senses. I'd be happy to make the name more specific.